### PR TITLE
[MNG-7390] Testing whether modules can be selected outside the cwd.

### PR DESCRIPTION
**Pull request by MartinKanters**
Relevant Maven core change: https://github.com/apache/maven/pull/677

### CODE REVIEW

**Comment by mthmulders** [comment]
This one is essentially the same one as #135, isn't it?

**Comment by MartinKanters** [comment]
> This one is essentially the same one as #135, isn't it?

It is exactly the same one, but since I changed the name of the branch, GH automatically closed that PR.
Somehow I cannot change the target branch and reopen the PR.

**Comment by mthmulders** [comment]
> It is exactly the same one, but since I changed the name of the branch, GH automatically closed that PR.

Thanks for the confirmation!

**Review by mthmulders — APPROVED** [review]

**Comment by michael-o** [comment]
Will check

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    private File moduleADir;
+
+    public MavenITmng7390SelectModuleOutsideCwdTest()
+    {
+        super( "[4.0.0-alpha-1,)" );
+    }
+
+    @Override
+    protected void setUp() throws Exception {
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java:42]: Curly on new line

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        moduleADir = ResourceExtractor.simpleExtractResources( getClass(), "/mng-7390-pl-outside-cwd/module-a");
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java:43]: padding space at end

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        // Clean up target files from earlier runs (verifier.setAutoClean does not work, as we are reducing the reactor)
+        final Verifier verifier = newVerifier( moduleADir.getAbsolutePath() );
+        verifier.addCliOption( "-f" );
+        verifier.addCliOption( ".." );
+        verifier.executeGoal( "clean" );
```
Comment (michael-o) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java:49]: Is the setup already a test to test the traversal up the hierarchy?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java (hunk 2 above)
Comment (MartinKanters) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java:43]: Nice finds. I should configure my formatter correctly..

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java (hunk 1 above)
Comment (MartinKanters) [reply core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7390SelectModuleOutsideCwdTest.java:42]: Nice one, I also fixed the one at the class declaration.

**Pull request closed without merging**